PWGCF/FemtoDream/TableProducer/femtoDreamProducerTask.cxx

```diff
@@ -513,6 +564,204 @@ struct femtoDreamProducerTask {
     }
   }
 
+  template <bool isMC, typename TrackType>
+  bool fillTracksForCharmHadron(TrackType const& tracks, o2::aod::FemtoHFTrack const& prong0, o2::aod::FemtoHFTrack const& prong1, o2::aod::FemtoHFTrack const& prong2, int candSize)
+  {
+
+    std::vector<int> childIDs = {0, 0}; // these IDs are necessary to keep track of the children
+    // std::vector<int> tmpIDtrack;        // this vector keeps track of the matching of the primary track table row <-> aod::track table global index
+    bool fIsTrackFilled = false;
+
+    for (auto& track : tracks) {
+      /// if the most open selection criteria are not fulfilled there is no
+      /// point looking further at the track
+      if (!trackCuts.isSelectedMinimal(track)) {
+        continue;
+      }
+
+      if ((candSize == 1) && (track.globalIndex() == prong0.globalIndex() || track.globalIndex() == prong1.globalIndex() || track.globalIndex() == prong2.globalIndex()))
+        continue;
```

Comment:
This condition is only applicable if there is only one Lc candidate in a collision. So I think it will not have any harm but can reduce the size of derived data.